common/base.scm

```diff
@@ -264,7 +278,7 @@ The secondary value indicates the absence of an entry."
   (if (and (list? x)
            (eq? (car x) 'quote))
       (cond ((string? (cadr x))
-             (cadr x))
+             (read-from-string (cadr x)))
```

Comment:
I'm not really familiar with the code, but why should unquote-string return something different than a string?